src/mongocxx/collection.cpp

```diff
@@ -197,20 +196,20 @@ void* collection::implementation() const {
     return _impl->collection_t;
 }
 
-stdx::optional<result::insert_one> collection::insert_one(bsoncxx::document::view document,
+stdx::optional<result::insert_one> collection::insert_one(bsoncxx::document::view_or_value document,
                                                           const options::insert& options) {
     class bulk_write bulk_op(false);
     bsoncxx::document::element oid{};
 
     bsoncxx::builder::stream::document new_document;
-    if (!document["_id"]) {
+    if (!document.view()["_id"]) {
```

Comment:
Good point! Also this isn't really user-facing, is it? If not, it doesn't make much of a difference.